tests/src/Api/FileIngestionApiTest.php

```diff
@@ -26,14 +29,47 @@ class FileIngestionApiTest extends TestCase
      */
     public function testFileIngestion(array $clientConfig, array $responses, $expectedResult): void
     {
-        $actualResult = $this->getTestingClient($clientConfig, $responses)
+        $actualResult = $this->getTestingClient($clientConfig, $responses, [$this, 'inspectRequest'])
             ->ingestFile(
                 'http://example.com',
-                file_get_contents(__DIR__ . '/../../fixtures/files/image.png')
+                file_get_contents(__DIR__ . '/../../fixtures/files/image.png'),
+                ['en', 'ro'],
+                [
+                    'field1' => ['value1', 'value2'],
+                    'field2' => ['value3', 2345],
+                ],
+                'unique-my-ID',
+                ['user001', 'user02'],
+                ['user003', 'user04'],
+                ['group001', 'group002'],
+                ['group003', 'group004']
             );
         $this->assertEquals($expectedResult, $actualResult);
     }
 
+    /**
+     * @param RequestInterface $request
+     */
+    public function inspectRequest(RequestInterface $request): void
+    {
+        if ($request->getUri() == 'http://example.com/token') {
+            $this->inspectTokenRequest($request);
+        } else {
+            $this->assertEquals('http://example.com/ingest?apiKey=bananas&database=cucumbers&uri=http%3A%2F%2Fexample.com&language=%5B%22en%22%2C%22ro%22%5D&reference=unique-my-ID', $request->getUri());
+            $this->inspectAuthorizationHeaders($request);
+            $this->inspectBoundary($request);
+            $this->inspectBoundary($request);
+            $parts = $this->getMultiParts($request);
```

Comment:
After getting the parts, let's assert the number. For example a bug in metadata manipulation inserted empty metadata in the body. Such an assertion could catch such issues.